### PR TITLE
[WIP] Add automatic replacement of dead nodes

### LINKED ISSUE #507: Automatically replace scylla node that looses storage
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
When a node is decommissioned, the local storage is lost as well. Currently it requires a manual action by annotating the service to trigger replacement, otherwise the new pod is stuck on join as it doesn't have replace-address-first-boot set.

**What is use case behind this feature:**
Stability - operator should be able to run scylla without any user intervention.

**Additional Information:**
One option is to write down a file in an init container, check its presents and generate the replace-address-first-boot. Maybe there is a more sophisticated way to get the same information directly from scylla.

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:**
- Add an initContainer detecting dead nodes needing replacement
- Add setting "replace-address-first-boot" flag for auto-detected dead nodes in sidecar
- Add an e2e tests for auto-replacement of dead nodes

**Which issue is resolved by this Pull Request:**
Resolves #507

**Requires:**
- #721

### CODE REVIEW

**Comment by rzetelskik** [comment]
@zimnx @tnozicka Please review.

File: pkg/cmd/operator/auto_replace.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		return nil
+	}
+
+	// Create dummy cluster for helper functions
+	sc := &scyllav1.ScyllaCluster{ObjectMeta: metav1.ObjectMeta{
+		Name:      pod.Labels[naming.ClusterNameLabel],
+		Namespace: o.Namespace,
+	}}
+
+	scyllaClient, _, err := scyllaclient.GetScyllaClientForScyllaCluster(ctx, o.kubeClient.CoreV1(), sc)
```
Comment (tnozicka) [inline pkg/cmd/operator/auto_replace.go:161]: hm, if we are taking just to the other nodes we could do it from the operator

File: pkg/cmd/operator/auto_replace.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	for i := range podList.Items {
+		p := &podList.Items[i]
+		if p.Name != pod.Name {
+			foundOther = true
+			break
+		}
+	}
+	if !foundOther {
+		// First node in cluster skips bootstrap, can't replace other node
+		klog.Infof("first node in cluster, skipping")
```
Comment (tnozicka) [inline pkg/cmd/operator/auto_replace.go:151]: if there are 3 nodes and the first one looses storage, how does this help?

**Review by tnozicka — COMMENTED** [review]
I am not sure if this is the right approach or not, but I'd like to see a proposal first with different alternatives considered. Might save time on coding side.

**Comment by rzetelskik** [comment]
Closing this as it was submitted before considering #721. Will open a new PR if need be.

**Pull request closed without merging**